site/app/models/Notification.php

```diff
@@ -107,26 +107,23 @@ public static function createViewOnlyNotification($core, $details) {
     /**
      * Returns the corresponding url based on metadata
      *
-     * @param  Core     core
-     * @param  string   metadata
-     * @return string   url
+     * @param  string   $metadata_json
+     * @return string   $url
      */
-    public static function getUrl($core, $metadata_json) {
+    public static function getUrl($metadata_json) {
         $metadata = json_decode($metadata_json);
         if(is_null($metadata)) {
             return null;
         }
-        $parts = $metadata[0];
-        $hash = $metadata[1] ?? null;
-        return $core->buildUrl($parts, $hash);
+        return $metadata->url;
     }
 
     public static function getThreadIdIfExists($metadata_json) {
         $metadata = json_decode($metadata_json, true);
         if(is_null($metadata)) {
             return null;
         }
-        $thread_id = array_key_exists('thread_id', $metadata[0]) ? $metadata[0]['thread_id'] : -1;
+        $thread_id = property_exists($metadata, 'thread_id') ? $metadata->thread_id : -1;
         return $thread_id;
     }
```

Comment:
Use a consistent group of parameters for `json_decode` for these (and other) functions. We mostly use true elsewhere in the codebase.

You can also simplify the check here with the null coalescing operator:
```
$thread_id = $metadata->thread_id ?? -1;
```